docs/cookbook/configuration.rst

```diff
@@ -49,6 +52,15 @@ With this config running:
 will now put the spec in ``spec/MarkdownSpec.php`` and the class will be created
 in  ``src/Markdown.php``.
 
+Alternatively, you can choose to use Composer to provide the necessary
+configuration:
+
+.. code-block:: yaml
+
+    namespace_providers:
```

Comment:
This could be more user-friendly - I think the concept of a namespace provider is pretty internal.

Maybe a simple boolean here?

Reply:
I'd like this behaviour to be the default in PhpSpec 5 so it's important we have a syntax that supports negation too.

Therefore maybe a boolean like `detect_composer_namespaces` is enough?

Reply:
> Therefore maybe a boolean like detect_composer_namespaces is enough?

I'm afraid it's not, because you can't pass configuration to the namespace provider with a boolean, which means you cannot customize the `root_directory` or the `spec_prefix` arguments.

Reply:
I forgot to document this BTW, it seems, so let's do it